niriPipe/inttests.py

```diff
@@ -155,4 +156,124 @@ def downloader_inttest():
     else:
         raise RuntimeError(
             "Hash of downloads '{}' is different ".format(out_hash) +
-            "than the expected value of '{}'.".format(expected_hash))
+            "than the expected value of '0d6ae285dcdb60904561aca79618afef'.")
+
+
+def finder_inttest():
+    """
+    Make sure the Finder class can find NIRI data.
+    """
+
+    """
+    First test: find data for stack GN-2019A-FT-108-12.
+
+    Stack name:  GN-2019A-FT-108-12
+    Object:      N20190405S01[11-37]
+    Flats:       N20190406S00[07-26]
+    Shortdark:   N20190406S01[12-21]
+    Longdarks:   N20190406S00[42-56]
+    """
+
+    state = {
+        'config': {
+            'DATAFINDER': {
+                'min_objects': 1,
+                'min_flats': 1,
+                'min_longdarks': 1,
+                'min_shortdarks': 0
+            }
+        },
+        'current_stack': {
+            'obs_name': 'GN-2019A-FT-108-12'
+        }
+    }
+
+    module_logger.info("Starting finder inttest with stack {}".format(
+        state['current_stack']['obs_name']))
+
+    CADC_data_finder = niriPipe.utils.finder.Finder(state=state)
+    table = CADC_data_finder.run()
+    table.sort(['productID'])
+
+    module_logger.debug("Resulting table:")
+    module_logger.debug('\n'+'\n'.join(table.pformat_all()))
+
+    # Object frames
+    desired_frames = \
+        ['N20190405S01'+str(x) for x in range(11, 38)]
+    # Flats
+    desired_frames.extend(
+        ['N20190406S00'+'{:02d}'.format(x) for x in range(7, 27)])
+    # Short darks
+    desired_frames.extend(
+        ['N20190406S01'+'{:02d}'.format(x) for x in range(12, 22)])
+    # Long darks
+    desired_frames.extend(
+        ['N20190406S00'+'{:02d}'.format(x) for x in range(42, 57)])
+
+    desired_frames.sort()
+
+    for frame in desired_frames:
+        if frame not in table['productID']:
```

Comment:
This could also be restated as :
```
assert frame in table('productID'], "Frame {} not found.".format(frame)
```

This is consistent with the approach taken for the unit tests, although the execution result is `AssertionError`, not `RuntimeError`. This comment is purely a style comment, so do with it as you wish.

Reply:
Done, actually, I didn't know of this use of assert. Great to know!